tests/unit/modules/test_boto3_route53.py

```diff
@@ -0,0 +1,153 @@
+from tests.support.mock import MagicMock, patch
+from tests.support.unit import TestCase, skipIf
+
+try:
+    import boto3
+
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+# the boto3_route53 module relies on the connect_to_region() method
+# which was added in boto 2.8.0
+# https://github.com/boto/boto/commit/33ac26b416fbb48a60602542b4ce15dcc7029f12
+REQUIRED_BOTO3_VERSION = "1.2.1"
+
+
+def __virtual__():
+    """
+    Returns True/False boolean depending on if Boto3 is installed and correct
+    version.
+    """
+    if not HAS_BOTO3:
+        return False
+    if LooseVersion(boto3.__version__) < LooseVersion(REQUIRED_BOTO3_VERSION):
+        return (
+            False,
+            "The boto3 module must be greater or equal to version {}".format(
+                REQUIRED_BOTO3_VERSION
+            ),
+        )
+    return True
+
+
+REGION = "us-east-1"
+ACCESS_KEY = "GKTADJGHEIQSXMKKRBJ08H"
+SECRET_KEY = "askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs"
+CONN_PARAMETERS = {
+    "region": REGION,
+    "key": ACCESS_KEY,
+    "keyid": SECRET_KEY,
+    "profile": {},
+}
+
+LIST_RESOURCE_RECORD_SETS_RETURN = {
+    "IsTruncated": True,
+    "MaxItems": "100",
+    "NextRecordName": "blog3.saltstack.furniture.",
+    "NextRecordType": "CNAME",
+    "ResourceRecordSets": [
+        {
+            "Name": "blog.saltstack.furniture.",
+            "ResourceRecords": [{"Value": "127.0.0.1"}],
+            "TTL": 60,
+            "Type": "A",
+        },
+        {
+            "Name": "blog2.saltstack.furniture.",
+            "ResourceRecords": [{"Value": "127.0.0.1"}],
+            "TTL": 60,
+            "Type": "A",
+        },
+    ],
+}
+
+
+@skipIf(HAS_BOTO3 is False, "The boto module must be installed.")
+@skipIf(
+    LooseVersion(boto3.__version__) < LooseVersion(REQUIRED_BOTO3_VERSION),
+    "The boto3 module must be greater or equal to version {}".format(
+        REQUIRED_BOTO3_VERSION
+    ),
+)
+class Boto3Route53TestCase(TestCase, LoaderModuleMockMixin):
+    """
+    TestCase for salt.modules.boto3_route53 moodule
+    """
+
+    conn = None
+
+    def setup_loader_modules(self):
+        self.opts = salt.config.DEFAULT_MINION_OPTS.copy()
+        utils = salt.loader.utils(
+            self.opts,
+            whitelist=["boto3", "args", "systemd", "path", "platform"],
+            context={},
+        )
+        return {boto3_route53: {"__utils__": utils}}
+
+    def setUp(self):
+        super().setUp()
+        boto3_route53.__init__(self.opts)
+        del self.opts
+
+        # Set up MagicMock to replace the boto3 session
+        # connections keep getting cached from prior tests, can't find the
+        # correct context object to clear it. So randomize the cache key, to prevent any
+        # cache hits
+        CONN_PARAMETERS["key"] = "".join(
+            random.choice(string.ascii_lowercase + string.digits) for _ in range(50)
+        )
+
+        self.conn = MagicMock()
```

Comment:
**suggestion(non-blocking)** `create_autospec` is preferred (or `patch(..., autospec=True)`) to a bare MagicMock, since it will get you the correct signature. Sometimes it's not appropriate, though.